github-data/src/main/java/com/example/github_data/repository/remote/RemoteGithubDataSourceImpl.kt

```diff
@@ -0,0 +1,26 @@
+package com.example.github_data.repository.remote
+
+import com.example.github_data.GitHubApi
+import com.example.github_data.GitHubService
+import com.example.github_data.room.GithubRepoEntity
+import kotlinx.coroutines.DelicateCoroutinesApi
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.async
+
+internal class RemoteGithubDataSourceImpl(githubService: GitHubApi? = null): RemoteGithubDataSource {
+    // retrofit통신으로 데이터 받아옴.
+    private val gitHubService = githubService ?: GitHubService.getGitHubService()
```

Comment:
GitHubApi를 nullable한 객체로 받아올필요가 있을까요?
+ 
생성시점에 주입받을수 있는구조를 사용해보아요! 
RemoteGithubDataSourceImpl는 GitHubService에 강한 의존이 가진 코드는 아닐까요?
(싱글톤 관련 코멘트 남긴부분 참고해보아요!)

```suggestion
internal class RemoteGithubDataSourceImpl(private val githubService: GitHubApi = GitHubService.getGitHubService() ): RemoteGithubDataSource {
```

Reply:
GitHubApi를 nullable한 객체로 받아왔던 건 테스트를 위해서였습니다. 하지만 조언해주신 방법으로 구성해도 테스트 코드에 지장이 없음을 확인하여 수정했습니다.